server/src/main/java/org/opensearch/rest/action/admin/indices/RestViewAction.java

```diff
@@ -0,0 +1,200 @@
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.ClusterStateUpdateTask;
+import org.opensearch.cluster.metadata.Metadata;
+import org.opensearch.cluster.metadata.View;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.ValidationException;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.BytesRestResponse;
+import org.opensearch.rest.NamedRoute;
+import org.opensearch.rest.RestChannel;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestResponse;
+import org.opensearch.rest.action.RestCancellableNodeClient;
+import org.opensearch.rest.action.RestStatusToXContentListener;
+import org.opensearch.rest.action.RestToXContentListener;
+import org.opensearch.rest.action.search.RestSearchAction;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.IntConsumer;
+import java.util.stream.Collectors;
+
+import static org.opensearch.rest.RestRequest.Method.DELETE;
+import static org.opensearch.rest.RestRequest.Method.GET;
+import static org.opensearch.rest.RestRequest.Method.POST;
+import static org.opensearch.rest.RestRequest.Method.PUT;
+
+/** All rest handlers for view actions */
+public class RestViewAction {
+
+    private final static Logger LOG = LogManager.getLogger(RestViewAction.class);
+
+    public static final String VIEW_ID = "view_id";
+    public static final String VIEW_ID_PARAMETER = "{" + VIEW_ID + "}";
+
+    /** Handler for create view */
+    public static class CreateViewHandler extends BaseRestHandler {
+
+        @Override
+        public List<Route> routes() {
+            return List.of(new NamedRoute.Builder().path("/views").method(POST).uniqueName(CreateViewAction.NAME).build());
+        }
+
+        @Override
+        public String getName() {
+            return CreateViewAction.NAME;
+        }
+
+        @Override
+        protected RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
+            try (final XContentParser parser = request.contentParser()) {
+                final CreateViewAction.Request createViewAction = CreateViewAction.Request.fromXContent(parser);
+                return channel -> client.admin().indices().createView(createViewAction, new RestToXContentListener<>(channel));
+            }
+        }
+    }
+
+    public static class SearchViewHandler extends BaseRestHandler {
+        @Override
+        public List<Route> routes() {
+            return List.of(
+                new NamedRoute.Builder().path("/views/" + VIEW_ID_PARAMETER + "/_search").method(GET).uniqueName("cluster:views:search").build(),
```

Comment:
Why `cluster` for the name here? 

The search transport action is `indices:data/read/search`, should this name follow a similar pattern?